### ISSUE TITLE
Unable to access built-in Kibana Dashboards

### ISSUE THREAD

**Issue by wicknet** [StamusNetworks/SELKS#314]
Wonderful IDS software. After installation and setup, Threat hunter, Management, Evebox - all work straight out of the box. But despite all attempts, cannot get Kibana to show the 27 built in dashboards that come bundled with SELKS.

I do have have extensive experience using Zeek (with filebeat to ingest Zeek logs) and Kibana. Filebeat is not installed in Selks, and I cannot find a Moloch ingest. Since logstash data cannot be ingest directly w/o filebeat in Kibana, not sure how I can get dashboards to show up? Only Overview section shows Logstash_* views, which then have to be selected individually and don't appear to be easily navigable.

Any help is deeply appreciated.

**Comment by pevma** [comment]
Thank you for trying it out. It provides much more than IDS alerts out of the box too.   
Please see attached - if you open Kibana or click on the Dashboards widget (right upper corner) - you should select the Dashboards menu and they should all appear. Is that the case? 
![Screenshot from 2021-05-17 14-13-07](https://user-images.githubusercontent.com/1405917/118486894-6c5d0400-b71a-11eb-8e34-fda4aa5bcc42.png)

**Comment by wicknet** [comment]
Thank you for your prompt response. Unfortunately, I am not seeing them as you are. Also, my webpage sage Analytics instead of Kibana. Please see attached pics of what I see. I've also included a screenshot of /etc/scirius/local_settings.py showing Kibana Dashboards. Have updated them as well from github.

<img width="1037" alt="Local_Settings" src="https://user-images.githubusercontent.com/18508394/118500487-d2836000-b6f5-11eb-97a9-f401462d4143.png">
<img width="1439" alt="Overview" src="https://user-images.githubusercontent.com/18508394/118500491-d2836000-b6f5-11eb-9051-b5613b24df94.png">
<img width="354" alt="Analytics" src="https://user-images.githubusercontent.com/18508394/118500495-d31bf680-b6f5-11eb-9f80-0a9aede07ada.png">
<img width="1439" alt="Dashboard" src="https://user-images.githubusercontent.com/18508394/118500496-d31bf680-b6f5-11eb-96b0-2dd78f8b5118.png">

**Comment by pevma** [comment]
Did the first time setup steps complete successfully  -  https://github.com/StamusNetworks/SELKS/wiki/First-time-setup  ?

**Comment by wicknet** [comment]
Yes. No errors. Ran it a few times to make sure. Passes all health checks too. Ran update script as well. Also tried resetting dashboards as mentioned in the Wiki.

**Comment by wicknet** [comment]
Output of Health Checks
root@SELKS:~# systemctl status suricata elasticsearch logstash kibana evebox molochviewer-selks molochpcapread-selks
● suricata.service - LSB: Next Generation IDS/IPS
   Loaded: loaded (/etc/init.d/suricata; generated)
   Active: active (running) since Mon 2021-05-17 09:32:52 EDT; 27min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 2163 ExecStart=/etc/init.d/suricata start (code=exited, status=0/SUCC
    Tasks: 14 (limit: 4915)
   Memory: 3.4G
   CGroup: /system.slice/suricata.service
           └─2216 /usr/bin/suricata -c /etc/suricata/suricata.yaml --pidfile /va

May 17 09:32:52 SELKS systemd[1]: Starting LSB: Next Generation IDS/IPS...
May 17 09:32:52 SELKS suricata[2163]: Starting suricata in IDS (af-packet) mode.
May 17 09:32:52 SELKS systemd[1]: Started LSB: Next Generation IDS/IPS.

● elasticsearch.service - Elasticsearch
   Loaded: loaded (/lib/systemd/system/elasticsearch.service; enabled; vendor pr
   Active: active (running) since Mon 2021-05-17 09:10:31 EDT; 50min ago
     Docs: https://www.elastic.co
 Main PID: 724 (java)
    Tasks: 120 (limit: 4915)
   Memory: 19.8G
   CGroup: /system.slice/elasticsearch.service
           ├─ 724 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Des.networ
root@SELKS:~# systemctl status suricata elasticsearch logstash kibana evebox molochviewer-selks molochpcapread-selks
● suricata.service - LSB: Next Generation IDS/IPS
   Loaded: loaded (/etc/init.d/suricata; generated)
   Active: active (running) since Mon 2021-05-17 09:32:52 EDT; 28min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 2163 ExecStart=/etc/init.d/suricata start (code=exited, status=0/SUCC
    Tasks: 14 (limit: 4915)
   Memory: 3.3G
   CGroup: /system.slice/suricata.service
           └─2216 /usr/bin/suricata -c /etc/suricata/suricata.yaml --pidfile /va

May 17 09:32:52 SELKS systemd[1]: Starting LSB: Next Generation IDS/IPS...
May 17 09:32:52 SELKS suricata[2163]: Starting suricata in IDS (af-packet) mode.
May 17 09:32:52 SELKS systemd[1]: Started LSB: Next Generation IDS/IPS.

● elasticsearch.service - Elasticsearch
   Loaded: loaded (/lib/systemd/system/elasticsearch.service; enabled; vendor pr
   Active: active (running) since Mon 2021-05-17 09:10:31 EDT; 50min ago
     Docs: https://www.elastic.co
 Main PID: 724 (java)
    Tasks: 120 (limit: 4915)
   Memory: 19.8G
   CGroup: /system.slice/elasticsearch.service
           ├─ 724 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Des.networ
           └─1063 /usr/share/elasticsearch/modules/x-pack-ml/platform/linux-x86_

May 17 09:09:50 SELKS systemd[1]: Starting Elasticsearch...
May 17 09:10:31 SELKS systemd[1]: Started Elasticsearch.

● logstash.service - logstash
   Loaded: loaded (/etc/systemd/system/logstash.service; enabled; vendor preset:
   Active: active (running) since Mon 2021-05-17 09:32:52 EDT; 28min ago
 Main PID: 2171 (java)
    Tasks: 55 (limit: 4915)
   Memory: 980.1M
   CGroup: /system.slice/logstash.service
           └─2171 /usr/share/logstash/jdk/bin/java -Xms1g -Xmx1g -XX:+UseConcMar

May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,048][INFO ][logstash.
May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,086][INFO ][logstash.
May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,087][INFO ][logstash.
May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,478][INFO ][logstash.
May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,737][INFO ][logstash.
May 17 09:33:29 SELKS logstash[2171]: [2021-05-17T09:33:29,849][INFO ][logstash.
May 17 09:33:32 SELKS logstash[2171]: [2021-05-17T09:33:32,131][INFO ][logstash.
May 17 09:33:32 SELKS logstash[2171]: [2021-05-17T09:33:32,483][INFO ][logstash.
May 17 09:33:32 SELKS logstash[2171]: [2021-05-17T09:33:32,583][INFO ][logstash.
May 17 09:33:32 SELKS logstash[2171]: [2021-05-17T09:33:32,586][INFO ][filewatch

● kibana.service - Kibana
   Loaded: loaded (/etc/systemd/system/kibana.service; enabled; vendor preset: e
   Active: active (running) since Mon 2021-05-17 09:09:50 EDT; 51min ago
     Docs: https://www.elastic.co
 Main PID: 727 (node)
    Tasks: 11 (limit: 4915)
   Memory: 554.2M
   CGroup: /system.slice/kibana.service
           └─727 /usr/share/kibana/bin/../node/bin/node /usr/share/kibana/bin/..

May 17 09:09:50 SELKS systemd[1]: Started Kibana.

● evebox.service - EveBox Server
   Loaded: loaded (/lib/systemd/system/evebox.service; enabled; vendor preset: e
   Active: active (running) since Mon 2021-05-17 09:09:49 EDT; 51min ago
 Main PID: 538 (evebox)
    Tasks: 9 (limit: 4915)
   Memory: 12.9M
   CGroup: /system.slice/evebox.service
           └─538 /usr/bin/evebox server

May 17 09:09:49 SELKS systemd[1]: Started EveBox Server.
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::version: Th
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::server::mai
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::sqlite::ini
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::sqlite::ini
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::sqlite::ini
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49 ERROR evebox::server::mai
May 17 09:09:49 SELKS evebox[538]: 2021-05-17 09:09:49  INFO evebox::server::mai

● molochviewer-selks.service - Moloch Viewer
   Loaded: loaded (/etc/systemd/system/molochviewer-selks.service; enabled; vend
   Active: active (running) since Mon 2021-05-17 09:32:52 EDT; 28min ago
 Main PID: 2178 (sh)
    Tasks: 12 (limit: 4915)
   Memory: 41.5M
   CGroup: /system.slice/molochviewer-selks.service
           ├─2178 /bin/sh -c /data/moloch/bin/node viewer.js -c /data/moloch/etc
           └─2180 /data/moloch/bin/node viewer.js -c /data/moloch/etc/config.ini

May 17 09:32:52 SELKS systemd[1]: Started Moloch Viewer.

● molochpcapread-selks.service - Moloch Pcap Read
   Loaded: loaded (/etc/systemd/system/molochpcapread-selks.service; enabled; ve
   Active: active (running) since Mon 2021-05-17 09:32:52 EDT; 28min ago
 Main PID: 2193 (sh)
    Tasks: 6 (limit: 4915)
   Memory: 436.0M
   CGroup: /system.slice/molochpcapread-selks.service
           ├─2193 /bin/sh -c /data/moloch/bin/moloch-capture -c /data/moloch/etc
           └─2195 /data/moloch/bin/moloch-capture -c /data/moloch/etc/config.ini

May 17 09:32:52 SELKS systemd[1]: Started Moloch Pcap Read.

**Comment by pevma** [comment]
Actually - these here  that you edited - should stays as they were defaults  by the package - https://github.com/StamusNetworks/SELKS/issues/314#issuecomment-842344444  

Can you post the last few lines from the health check ? It seems they are missing.

You can also reset the dashboards via the gui - https://github.com/StamusNetworks/SELKS/wiki/How-to-load-or-update-dashboards#from-scirius

**Comment by wicknet** [comment]
Without adding: KIBANA7_DASHBOARDS_PATH = "/opt/selks/kibana7-dashboards/" in etc/scirius/local_settings.py, the first time setup fails consistently with error: 

AttributeError: 'Settings' object has no attribute 'KIBANA7_DASHBOARDS_PATH'
Dashboards loading set up job failed...Exiting.

Only after adding that line did the first time setup exit normally.

Latest Health_Check attached as txt file

[Health_Check.txt](https://github.com/StamusNetworks/SELKS/files/6495603/Health_Check.txt)

**Comment by pevma** [comment]
ok - can you try to adjust the dashboards line to be exactly that
`KIBANA6_DASHBOARDS_PATH = "/opt/selks/kibana7-dashboards/"`

Then form the GUI in Scirius 
Go to System settings (from the Stamus logo drop down menu in the left upper corner) -> Kibana -> choose to reset the dashboards.

**Comment by wicknet** [comment]
Just did a fresh reinstall with a newly downloaded no-desktop iso . VM assigned 8 vCPU, 32Gb RAM and 500Gb HDD with 2 NICs, one of which is set in promiscuous mode on physical and vSwitch. 

After installation, selks-first-time-setup_stamus script fails with exact same error as before - AttributeError: 'Settings' object has no attribute 'KIBANA7_DASHBOARDS_PATH'
Dashboards loading set up job failed...Exiting...
### Exited with ERROR  ###

/etc/scirius/local_settings.py file does have the line exactly as suggested by you:

KIBANA6_DASHBOARDS_PATH = "/opt/selks/kibana7-dashboards/"

Attaching both files for your review.  Please suggest any other troubleshooting I can do to get this running.  THANKS!

[settings.txt](https://github.com/StamusNetworks/SELKS/files/6497050/settings.txt)
[clean-setup-error.txt](https://github.com/StamusNetworks/SELKS/files/6497051/clean-setup-error.txt)

**Comment by wicknet** [comment]
Also, as I'm sure you know, on a default clean install, Dashboards show just a white background with no menus until a different config is loaded for nginx proxy from Github and it's restarted. Just fixed that now. However, first-time-setup still fails on the Kibana issue.

**Comment by wicknet** [comment]
Issue solved. I recreated the VM again but this time, without updating the underlying Debian OS first, I ran the selks-first-setup which ran without issues. Than ran selks-upgrade and Kibana dashboard came up fine and showed the logstash-* indices as default with all 27 dashboards available. 

Only after that I ran apt-update and apt-upgrade and nothing broke.

Many thanks for all the troubleshooting help. 

Loving the dashboards - SO informative! Kudos on a great job with SELKS.
